File: llm_generated_code/ChatGPT/CPP/Level1/Q8.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct Meeting {
    int start;
    int end;
    int index;
};

bool compare(Meeting m1, Meeting m2) {
    // Sort by end time, and if equal, by index
    if (m1.end == m2.end)
        return m1.index < m2.index;
    return m1.end < m2.end;
}

vector<int> maxMeetings(vector<int>& start, vector<int>& end) {
    int n = start.size();
    vector<Meeting> meetings;

    // Create meeting objects with start, end, and index
    for (int i = 0; i < n; i++) {
        meetings.push_back({start[i], end[i], i + 1});
    }

    // Sort meetings by end time (and by index if end times are equal)
    sort(meetings.begin(), meetings.end(), compare);

    vector<int> result;
    int lastEnd = -1;

    // Select meetings that don't overlap
    for (const auto& m : meetings) {
        if (m.start > lastEnd) {
            result.push_back(m.index);
            lastEnd = m.end;
        }
    }

    return result;
}

int main() {
    vector<int> start = {1, 3, 0, 5, 8, 5};
    vector<int> end = {2, 4, 6, 7, 9, 9};

    vector<int> order = maxMeetings(start, end);

    cout << "Maximum number of meetings that can be accommodated: " << order.size() << endl;
    cout << "Order of meetings: ";
    for (int i : order) {
        cout << i << " ";
    }
    cout << endl;

    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q8.py
def max_meetings(start, end):
    # Create a list of tuples where each tuple contains the start and end time of meetings along with the original index
    meetings = sorted((e, s, i) for i, (s, e) in enumerate(zip(start, end)))
    
    # Initialize the variables
    max_meetings_list = []  # To store the order of meetings
    last_end_time = 0  # To track the end time of the last added meeting

    # Iterate through the sorted list of meetings
    for end_time, start_time, index in meetings:
        # Check if the meeting can be attended by comparing its start time with the last attended meeting's end time
        if start_time >= last_end_time:
            max_meetings_list.append(index + 1)  # Store the original index (1-based)
            last_end_time = end_time  # Update the last_end_time to the current meeting's end time

    return max_meetings_list

# Example usage
start_times = [1, 3, 0, 5, 8, 5]
end_times = [2, 4, 6, 7, 9, 9]
print(max_meetings(start_times, end_times))

